spec/features/applications/show_spec.rb

```diff
@@ -70,6 +70,22 @@
           expect(page).to have_content(@pet_3.name)
         end
       end
+
+      describe "Adopt this Pet" do
+        it "has an 'Adopt this Pet' button next to each Pet name that links back to the application show page" do
+          visit "/applications/#{@application.id}"
+          
+          expect(page).to_not have_content(@pet_3.name)
+          fill_in "Pet Search", with: @pet_3.name
+          click_button "Submit"
+          
+          expect(page).to have_button("Adopt this Pet")
+          click_button "Adopt this Pet"
+      
+          expect(page).to have_content(@pet_3.name)
+          expect(current_path).to eq("/applications/#{@application.id}")
+        end
+      end
```

Comment:
Need to add tests for
* When we click the button, it checks to see if pet_3 is now included in @application.pets (more than the page content)
* Cases when we type a name and multiple pets come up, not just one.
* For more specific "within" testing, do we want to add a unique <div> class or id to these elements?

Reply:
This was fixed using validation in the model.